packages/ui/src/components/TaskCard/TaskCard.tsx

```diff
@@ -0,0 +1,135 @@
+import { cloneElement, ReactElement, ReactNode } from "react";
+
+import { IconType } from "../../types";
+import { LinkProps } from "../Link";
+import { HeadingLevel } from "../Heading";
+import { headingStyles as headingComponentStyles } from "../Heading/styles";
+
+export const colorOptions = [
+  "primary",
+  "secondary",
+  "tertiary",
+  "quaternary",
+  "quinary",
+] as const;
+
+type CardColor = (typeof colorOptions)[number];
+
+const headingBarStyleMap: Record<CardColor, Record<string, string>> = {
+  primary: {
+    "data-h2-background-color": "base(primary.lightest)",
+    "data-h2-border-bottom": "base(primary.darkest)",
+  },
+  secondary: {
+    "data-h2-background-color": "base(secondary.lightest)",
+    "data-h2-border-bottom": "base(secondary.darkest)",
+  },
+  tertiary: {
+    "data-h2-background-color": "base(tertiary.lightest)",
+    "data-h2-border-bottom": "base(tertiary.darkest)",
+  },
+  quaternary: {
+    "data-h2-background-color": "base(quaternary.lightest)",
+    "data-h2-border-bottom": "base(quaternary.darkest)",
+  },
+  quinary: {
+    "data-h2-background-color": "base(quinary.lightest)",
+    "data-h2-border-bottom": "base(quinary.darkest)",
+  },
+};
+
+const wrapperStyleMap: Record<CardColor, Record<string, string>> = {
+  primary: {
+    "data-h2-color": "base(primary.darkest)",
+  },
+  secondary: {
+    "data-h2-color": "base(secondary.darkest)",
+  },
+  tertiary: {
+    "data-h2-color": "base(tertiary.darkest)",
+  },
+  quaternary: {
+    "data-h2-color": "base(quaternary.darkest)",
+  },
+  quinary: {
+    "data-h2-color": "base(quinary.darkest)",
+  },
+};
+
+export interface TaskCardProps {
+  icon?: IconType;
+  title: ReactNode;
+  headingColor?: CardColor;
+  link?: ReactElement<LinkProps>;
```

Comment:
I noticed that Storybook flags an a11y issue on colour contrast of the link - but only when you switch modes.  I'm not sure if that's legitimate?  Also, I don't want to create a custom style exception for the links here.  That always causes pain when the standard styles move.  I'm not sure what to do here.

Reply:
The standard link colour is <color>.darker while the design uses <color>.darkest.  The standard design reserves the darkest colour variant for hover.

Reply:
Axe jest doesn't seem to mind.  :shrug:

Reply:
Ya I'm also dealing with a color contrast issue on my component, which I'm not sure is legitimate either. Axe jest doesn't mind, as well as axe dev tools.

I checked the first elements contrast ratio using [WebAIM](https://webaim.org/resources/contrastchecker/) and it passed all the checks :shrug

![image](https://github.com/user-attachments/assets/eadeacbd-1fb9-41dc-ad63-29fb30e5c129)

Reply:
Ya I'm also dealing with a color contrast issue on my component, which I'm not sure is legitimate either. Axe jest doesn't mind, as well as axe dev tools.

I checked the first elements contrast ratio using [WebAIM](https://webaim.org/resources/contrastchecker/) and it passed all the checks :shrug

![image](https://github.com/user-attachments/assets/eadeacbd-1fb9-41dc-ad63-29fb30e5c129)